indico/modules/events/editing/client/js/management/editable_type/NextEditable.jsx

```diff
@@ -0,0 +1,273 @@
+      onClose={onClose}
+      fileTypes={fileTypes}
+    />
+  );
+}
+
+NextEditable.propTypes = {
+  eventId: PropTypes.number.isRequired,
+  editableType: PropTypes.string.isRequired,
+  onClose: PropTypes.func.isRequired,
+};
+
+function NextEditableDisplay({eventId, editableType, onClose, fileTypes}) {
+  const [filters, setFilters] = useState({});
+  const [filteredEditables, setFilteredEditables] = useState(null);
+  const [selectedEditable, setSelectedEditable] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    (async () => {
+      setSelectedEditable(null);
+      setLoading(true);
+      let response;
+      try {
+        response = await indicoAxios.post(editableListURL({confId: eventId, type: editableType}), {
+          extensions: _.pickBy(filters, x => Array.isArray(x)),
+          has_files: _.pickBy(filters, x => !Array.isArray(x)),
+        });
+      } catch (e) {
+        handleAxiosError(e);
+        setLoading(false);
+        return;
+      }
+      setFilteredEditables(camelizeKeys(response.data));
+      setLoading(false);
+    })();
+  }, [eventId, editableType, filters]);
+
+  const updatePresenceFilter = (id, present) => {
+    if (filters[id] === present) {
+      setFilters(_.omit(filters, id));
+    } else {
+      setFilters({...filters, [id]: present});
+    }
+  };
+
+  const updateExtensionFilter = (id, extension) => {
+    const typeFilter = filters[id];
+    if (!Array.isArray(typeFilter)) {
+      // switching from no filter or presence filter to extension filter
+      setFilters({...filters, [id]: [extension]});
+    } else if (!typeFilter.includes(extension)) {
+      // adding previously-eselected extension
+      setFilters({...filters, [id]: [...typeFilter, extension]});
+    } else if (typeFilter.length === 1) {
+      // deselecting last extension
+      setFilters(_.omit(filters, id));
+    } else {
+      // deselecting some extension
+      setFilters({...filters, [id]: _.without(typeFilter, extension)});
+    }
+  };
+
+  const handleAssign = async () => {
+    try {
+      await indicoAxios.put(
+        assignMyselfURL({
+          confId: eventId,
+          contrib_id: selectedEditable.contributionId,
+          type: editableType,
+        })
+      );
+    } catch (error) {
+      handleAxiosError(error);
+      return;
+    }
+    location.href = selectedEditable.timelineURL;
+  };
+
+  const titleByType = {
+    paper: Translate.string('Get next paper'),
+    poster: Translate.string('Get next poster'),
+    slides: Translate.string('Get next slides'),
+  };
+
+  return (
+    <Modal onClose={onClose} closeOnDimmerClick={false} open>
+      <Modal.Header>{titleByType[editableType]}</Modal.Header>
+      <Modal.Content>
+        <div styleName="filetype-list">
+          {fileTypes.map(fileType => {
+            return (
+              <div key={fileType.id} styleName="filetype-list-item">
+                <div>
```

Comment:
I think this `div` can be omitted here since it's plain text inside and doesn't need to add any extra css. I can also be replaced with a `span`.